wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -1094,18 +1095,15 @@ public final class SessionManager: NSObject, SessionManagerType {
                         )
                     }
 
-                    await MainActor.run {
-                        let userSession = self.startBackgroundSession(
-                            for: account,
-                            with: coreDataStack,
-                            journal: journal
-                        )
+                    let userSession = await self.startBackgroundSession(
+                        for: account,
+                        with: coreDataStack,
+                        journal: journal
+                    )
 
-                        self.triggerMigrationsNeedsActionsIfNeeded(
-                            journal: journal,
-                            userSession: userSession
-                        )
+                    await userSession.performAppMigrationsIfNeeded()
```

Comment:
issue: these migrations may take some time, so I think we should move the UI to the migration screen.